docs/sql-ref-literals.md

```diff
@@ -0,0 +1,385 @@
+
+SELECT 9223372036854775807l AS col;
+  +-------------------+
+  |                col|
+  +-------------------+
+  |9223372036854775807|
+  +-------------------+
+
+SELECT -32Y AS col;
+  +---+
+  |col|
+  +---+
+  |-32|
+  +---+
+
+SELECT 482S AS col;
+  +---+
+  |col|
+  +---+
+  |482|
+  +---+
+{% endhighlight %}
+
+#### Decimal Literal
+
+#### <em>Syntax</em>
+
+{% highlight sql %}
+ [ + | - ] { digit [ ... ] . [ digit [ ... ] ] | . digit [ ... ] }
+{% endhighlight %}
+
+#### <em>Examples</em>
+
+{% highlight sql %}
+SELECT 12.578 AS col;
+  +------+
+  |   col|
+  +------+
+  |12.578|
+  +------+
+
+SELECT -0.1234567 AS col;
+  +----------+
+  |       col|
+  +----------+
+  |-0.1234567|
+  +----------+
+
+SELECT -.1234567 AS col;
+  +----------+
+  |       col|
+  +----------+
+  |-0.1234567|
+  +----------+
+{% endhighlight %}
+
+#### Floating Point and BigDecimal Literals
+
+#### <em>Syntax</em>
+
+{% highlight sql %}
+ [ + | - ] { digit [ ... ] [ E [ + | - ] digit [ ... ] ] [ D | BD ]
+             | digit [ ... ] . [ digit [ ... ] ] [ E [ + | - ] digit [ ... ] ] [ D | BD ]
+             | . digit [ ... ] [ E [ + | - ] digit [ ... ] ] [ D | BD ] }
+{% endhighlight %}
+digit: one of 0, 1, 2, 3, 4, 5, 6, 7, 8 or 9. <br>
+default (no postfix): indicate a 4-byte single-precision floating point number.<br>
+D: case insensitive, indicates <code>DOUBLE</code>, which is a 8-byte double-precision floating point number.<br>
+BD: case insensitive, indicates <code>BIGDECIMAL</code>, which is an arbitrary-precision signed decimal number.<br>
+#### <em>Examples</em>
+
+{% highlight sql %}
+SELECT 5E2 AS col;
+  +-----+
+  |  col|
+  +-----+
+  |500.0|
+  +-----+
+
+SELECT 5D AS col;
+  +---+
+  |col|
+  +---+
+  |5.0|
+  +---+
+
+SELECT -5BD AS col;
+  +---+
+  |col|
+  +---+
+  |-5 |
+  +---+
+
+SELECT 12.578e-2d AS col;
+  +-------+
+  |    col|
+  +-------+
+  |0.12578|
+  +-------+
+
+SELECT -.1234567E+2BD AS col;
+  +---------+
+  |      col|
+  +---------+
+  |-12.34567|
+  +---------+
+
+SELECT +3.e+3 AS col;
+  +------+
+  |   col|
+  +------+
+  |3000.0|
+  +------+
+
+SELECT -3.E-3D AS col;
+  +------+
+  |   col|
+  +------+
+  |-0.003|
+  +------+
+{% endhighlight %}
+
+### Datetime Literal
+
+A Datetime literal is used to specify a datetime value.
+
+#### Date Literal
+
+#### <em>Syntax</em>
+
+{% highlight sql %}
+DATE 'yyyy [ -MM-dd ]'
+{% endhighlight %}
+
+#### <em>Examples</em>
+
+{% highlight sql %}
+SELECT DATE '1997' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |1997-01-01|
+  +----------+
+
+SELECT TIMESTAMP '1997-01' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |1997-01-01|
+  +----------+
+
+SELECT DATE '2011-11-11' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |2011-11-11|
+  +----------+
+{% endhighlight %}
+
+#### Timestamp Literal
+
+#### <em>Syntax</em>
+
+{% highlight sql %}
+TIMESTAMP { 'yyyy [ -MM-dd HH:mm:ss.SSSSSSzzz ]' | '[ yyyy-MM-dd ] HH:mm [ :ss.SSSSSSzzz ]' }
```

Comment:
Here are the patterns https://github.com/apache/spark/blob/744c2480b580e0b68387328926ef7634cfb93adc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L257-L276

Reply:
@MaxGekk 
Thanks for the patterns. I tested on spark-sql and found that a few more (e.g. ```'[T][h]h:'```) also work. I list them all. Could you take a quick look to see if the list is OK?